src/dotnet/APIView/APIViewWeb/Managers/APIRevisionsManager.cs

```diff
@@ -425,6 +427,88 @@ public async Task<bool> IsAPIRevisionTheSame(APIRevisionListItemModel revision,
             return lastRevisionTextLines.SequenceEqual(fileTextLines);
         }
 
+        /// <summary>
+        /// Update APIRevision
+        /// </summary>
+        /// <param name="revision"></param>
+        /// <param name="languageService"></param>
+        /// <param name="telemetryClient"></param>
+        /// <returns></returns>
+        public async Task UpdateAPIRevisionAsync(APIRevisionListItemModel revision, LanguageService languageService, TelemetryClient telemetryClient)
+        {
+            foreach (var file in revision.Files)
+            {
+                if (!file.HasOriginal || !languageService.CanUpdate(file.VersionString))
+                {
+                    continue;
+                }
+
+                try
+                {
+                    var fileOriginal = await _originalsRepository.GetOriginalAsync(file.FileId);
+                    // file.Name property has been repurposed to store package name and version string
+                    // This is causing issue when updating review using latest parser since it expects Name field as file name
+                    // We have added a new property FileName which is only set for new reviews
+                    // All older reviews needs to be handled by checking review name field
+                    var fileName = file.FileName ?? file.Name;
+                    var codeFile = await languageService.GetCodeFileAsync(fileName, fileOriginal, false);
+                    await _codeFileRepository.UpsertCodeFileAsync(revision.Id, file.FileId, codeFile);
+                    // update only version string
+                    file.VersionString = codeFile.VersionString;
+                    await _apiRevisionsRepository.UpsertAPIRevisionAsync(revision);
+                }
+                catch (Exception ex)
+                {
+                    telemetryClient.TrackTrace("Failed to update revision " + revision.Id);
+                    telemetryClient.TrackException(ex);
+                }
+            }
+        }
+        /// <summary>
+        /// SoftDelete APIRevision if its not been updated after many months
+        /// </summary>
+        /// <param name="archiveAfterMonths"></param>
+        /// <returns></returns>
+        public async Task AutoArchiveAPIRevisions(int archiveAfterMonths)
+        {
+            var lastUpdatedDate = DateTime.Now.Subtract(TimeSpan.FromDays(archiveAfterMonths * 30));
+            var manualRevisions = await _apiRevisionsRepository.GetAPIRevisionsAsync(lastUpdatedOn: lastUpdatedDate, apiRevisionType:  APIRevisionType.Manual);
```

Comment:
We should archive pull request reviews here. Not manual reviews.

Reply:
Are you sure about this. It used to be Manual reviews https://github.com/Azure/azure-sdk-tools/blob/8b4943ac480d9cd3c699507d8cfb1221cd9c8a01/src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs#L509